engine/src/main/scala/cromwell/webservice/metadata/MetadataBuilderActor.scala

```diff
@@ -93,17 +63,22 @@ object MetadataBuilderActor {
        * Important note: Indexes are used for sorting purposes ONLY
        * An index of 2 DOES NOT guarantee that a value will be at index 2 in a list
        */
-        val innerList = IndexedJsonList(Vector(innerValue.withIndex(indexes.last)))
-        val list = indexes.init.foldRight(innerList)((index, acc) => {
-          IndexedJsonList(Vector(acc.withIndex(index)))
-        })
+        val list = innerValue match {
+           // Empty value in a list means empty list
+          case TimestampedEmptyJson(timestamp) => TimestampedJsList(Map.empty, timestamp)
+          case nonEmptyValue =>
+            val innerList = TimestampedJsList(TreeMap(indexes.last.toInt -> innerValue), nonEmptyValue.timestamp)
+            indexes.init.foldRight(innerList)((index, acc) => {
+              TimestampedJsList(TreeMap(index.toInt -> acc), acc.timestamp)
+            })
```

Comment:
could you please drop a comment as to what this is doing?  I can't figure it out.  😦